syntax = "proto3";

package test.fuzz;

import "envoy/config/core/v3/base.proto";
import "envoy/config/core/v3/address.proto";

import "google/protobuf/wrappers.proto";

import "validate/validate.proto";

// Common fuzzing input types.

message Headers {
  repeated envoy.config.core.v3.HeaderValue headers = 1;
}

message StreamInfo {
  envoy.config.core.v3.Metadata dynamic_metadata = 1;
  uint64 start_time = 2;
  google.protobuf.UInt32Value response_code = 3;
  envoy.config.core.v3.Metadata upstream_metadata = 4;
  string requested_server_name = 5;
  envoy.config.core.v3.Address address = 6;
  envoy.config.core.v3.Address upstream_local_address = 7;
}
